developer_docs/semantic_stage.md

```diff
@@ -0,0 +1,180 @@
+# Semantic Stage
+
+The semantic stage is described by the file [pyccel.parser.semantic](../pyccel/parser/semantic.py).
+
+The semantic stage serves several purposes:
+1.  [**Types**](#Types) : Determine the type of each symbol
+2.  [**Impose Restrictions**](#Impose-restrictions) : Ensure that the code follows any restictions that Pyccel imposes
+3.  [**Function Recognition**](#Function-recognition) : Identify any functions that are recognised by Pyccel
+4.  [**Imports**](#Imports) : Identify imports
+5.  [**Low-level Objects**](#Low-level-objects) : Create any objects which are hidden in high-level code, but must appear explicitly in low-level code
+6.  [**Object Tree**](#Object-tree) : Ensure the object tree is correctly constructed
+7.  [**Name Collisions**](#Name-collisions) : Ensure any potential name collisions are avoided
+
+## Navigation
```

Comment:
This subsection is not enumerated above in the several purposes.

Reply:
I don't consider navigation to be a purpose. The list above details the aims of the files not the following sections.  Do you think this section needs to be mentioned beforehand?

Reply:
I thought that you forgot to mention it in the list since it has the same structure as the purposes.

Reply:
Ok. Do you think it needs to be mentioned? Or are you happy to resolve this conversation

Reply:
I'm happy to resolve it. Excellent work tough !!